core/src/main/java/org/geysermc/geyser/registry/type/ItemMappings.java

```diff
@@ -163,6 +163,12 @@ public ItemDefinition getDefinition(int bedrockId) {
         return this.itemDefinitions.get(bedrockId);
     }
 
+    public ItemDefinition getDefinition(String bedrockIdentifier) {
+        return this.itemDefinitions.values().stream().filter(
+                itemDefinition -> itemDefinition.getIdentifier().equals(bedrockIdentifier))
+                .findFirst().orElse(null);
+    }
```

Comment:
I really do like streams but tbh, for just finding the first matching value, an iterator would be better.

Also, please annotate the method as Nullable.

`public ItemDefinition getDefinition(int bedrockId)` should probably be annotated as Nullable
`public ItemMapping getMapping(ItemData data)` should probably be annotated as NonNull